src/firmware_new/boards/frankie_v1/io/allegro_a3931_motor_driver.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+/**
+ * This file is an abstraction around the Allegro A3931 Motor Driver
+ */
+
+#include "firmware_new/boards/frankie_v1/Drivers/STM32H7xx_HAL_Driver/Inc/stm32h7xx_hal.h"
+#include "firmware_new/boards/frankie_v1/io/gpio_pin.h"
+#include "firmware_new/boards/frankie_v1/io/pwm_pin.h"
+
+typedef enum
+{
+    CLOCKWISE,
+    COUNTERCLOCKWISE
+} AllegroA3931MotorDriverDriveDirection;
+
+typedef struct AllegroA3931MotorDriver AllegroA3931MotorDriver_t;
+
+/**
+ * Create a Motor Driver
+ * @param pwm_pin The PWM pin to the driver. Note that this must be configured to
+ *                operate in the correct frequency range for the motor driver. Please
+ *                talk to Elec. to determine this.
+ * @param reset_pin A GPIO pin connected to the "reset" pin on the motor driver.
+ * @param coast_pin A GPIO pin connected to the "coast" pin on the motor driver.
+ * @param mode_pin A GPIO pin connected to the "mode" pin on the motor driver.
+ * @param direction_pin A GPIO pin connected to the "direction" pin on the motor driver.
+ * @param brake_pin A GPIO pin connected to the "brake" pin on the motor driver.
+ * @param esf_pin A GPIO pin connected to the "ESF" pin on the motor driver.
```

Comment:
Maybe throw in a description of what 'ESF' is here

Reply:
Changed to:
```
 * @param esf_pin A GPIO pin connected to the "ESF" pin on the motor driver.
 *                The "ESF" pin, when active, causes the motor driver to stop the
 *                motor if a short is detected.
```